pontoon/base/fluent.py

```diff
@@ -158,3 +159,53 @@ def get_simple_preview(content):
         tree = translation_ast.attributes[0]
 
     return serialize_value(tree.value)
+
+
+def is_plural_expression(expression):
+    from pontoon.base.models import Locale
+
+    CLDR_PLURALS = [c for _, c in Locale.CLDR_PLURALS]
+
+    if isinstance(expression, ast.SelectExpression):
+        return all(
+            isinstance(variant.key, ast.NumberLiteral)
+            or variant.key.name in CLDR_PLURALS
+            for variant in expression.variants
+        )
+
+    return False
+
+
+def create_locale_plural_variants(node, locale):
+    if not is_plural_expression(node):
+        return
+
+    numeric = []
+    source_plurals = {}
+    locale_plurals = []
+    default = None
+
+    for variant in node.variants:
+        key = variant.key
+        if isinstance(key, ast.NumberLiteral):
+            numeric.append(variant)
+        elif isinstance(key, ast.Identifier) and key.name in locale.cldr_plurals_list():
+            source_plurals[key.name] = variant
+        if variant.default:
+            default = variant
+
+    for plural in locale.cldr_plurals_list().split(", "):
+        if plural in source_plurals.keys():
+            variant = source_plurals[plural]
+        else:
+            variant = copy.deepcopy(default)
+            variant.default = False
+            variant.key.name = plural
+        locale_plurals.append(variant)
+
+    variants = numeric + locale_plurals
```

Comment:
Why are `numeric` and `locale_plurals` accumulated separately and then joined, rather than just `variants` being built directly?